sw/device/tests/sram_ctrl_smoketest.c

```diff
@@ -0,0 +1,75 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+// Write / Read small chunks of data to SRAM to test
+// basic functionality of sram_ctrl
+static void write_read_check(mmio_region_t base_addr_region,
+                             const char *sram_name) {
+  uint32_t rw_data_32;
+  for (int i = 0; i < 16; i++) {
+    rw_data_32 = (i << 26 | i << 19 | i << 11 | i << 3 | i >> 5);
+    mmio_region_write32(base_addr_region, i << 2, rw_data_32);
+  }
+  for (int i = 0; i < 16; i++) {
+    rw_data_32 = mmio_region_read32(base_addr_region, i << 2);
+    CHECK(rw_data_32 == (i << 26 | i << 19 | i << 11 | i << 3 | i >> 5),
+          "Memory Write/Read Mismatch for %s", sram_name);
```

Comment:
Consider recording the index that failed and the expected vs unexpected values, too.